pumps/syslog.go

```diff
@@ -0,0 +1,142 @@
+package pumps
+
+import (
+	"fmt"
+	"log/syslog"
+
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/TykTechnologies/logrus"
+	"github.com/TykTechnologies/tyk-pump/analytics"
+)
+
+type SyslogPump struct {
+	syslogConf *SyslogConf
+	writer     *syslog.Writer
+}
+
+var (
+	logPrefix = "syslog-pump"
+)
+
+type SyslogConf struct {
+	Transport   string `mapstructure:"transport"`
+	NetworkAddr string `mapstructure:"network_addr"`
+	LogLevel    int    `mapstructure:"log_level"`
+}
+
+func (s *SyslogPump) GetName() string {
+	return "Syslog Pump"
+}
+
+func (s *SyslogPump) New() Pump {
+	newPump := SyslogPump{}
+	return &newPump
+}
+
+func (s *SyslogPump) Init(config interface{}) error {
+	//Read configuration file
+	s.syslogConf = &SyslogConf{}
+	err := mapstructure.Decode(config, &s.syslogConf)
+
+	if err != nil {
+		log.WithFields(logrus.Fields{
+			"prefix": logPrefix,
+		}).Fatal("Failed to decode configuration: ", err)
+	}
+
+	// Init the configs
+	initConfigs(s)
+
+	// Init the Syslog writer
+	initWriter(s)
+
+	log.Debug("Syslog Pump active")
+	return nil
+}
+
+func initWriter(s *SyslogPump) {
+	syslogWriter, err := syslog.Dial(
+		s.syslogConf.Transport,
+		s.syslogConf.NetworkAddr,
+		syslog.Priority(s.syslogConf.LogLevel),
+		"syslog-pump")
```

Comment:
tag could be configurable and defaulting to `logPrefix` 's value

Reply:
Added `tag` to config options.